events/sink.go

```diff
@@ -0,0 +1,59 @@
+package events
+
+import (
+	"context"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containerd/containerd/api/types/event"
+	"github.com/containerd/containerd/namespaces"
+	goevents "github.com/docker/go-events"
+	"github.com/pkg/errors"
+)
+
+type sinkEvent struct {
+	ctx   context.Context
+	event Event
+}
+
+type eventSink struct {
```

Comment:
Could this be replaced by a combination of `events.Chanel` and `events.Filter`?

`events.Channel` gracefully handles closes when there are writes outstanding. It looks like this implementation wouldn't. If we can provide a guarantee that `Close` will never cross with a `Write`, it doesn't matter, but I'm not sure we can.